### ISSUE TITLE
Extend APIClient with Additional Constructor Attributes

### ISSUE THREAD

**Issue by bduzik** [MikeWooster/api-client#72]
**Is your feature request related to a problem? Please describe.**
Extending a the client class with new attributes doesn't work with paginators.  
I'd like to be able to extend the client class.

```python
class My_Client(APIClient):
    def __init__(self, base_url, **kwargs):
        self.base_url = base_url
        super().__init__(**kwargs)
```
```
Traceback (most recent call last):
  File "./example.py", line 1397, in <module>
    my_id=my_id))
  File ".venv/lib/python3.6/site-packages/apiclient/paginators.py", line 36, in wrap
    with set_strategy(client, strategy) as temporary_client:
  File "/usr/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File ".venv/lib/python3.6/site-packages/apiclient/paginators.py", line 17, in set_strategy
    temporary_client = client.clone()
  File ".venv/lib/python3.6/site-packages/apiclient/client.py", line 102, in clone
    request_formatter=self.get_request_formatter(),
TypeError: __init__() missing 1 required positional argument: 'base_url'
```

**Describe the solution you'd like**

In client.py replace clone function with deepcopy from the standard library. I've tested this an it works.

**Describe alternatives you've considered**
You can do this other ways.  I've extended the endpoint class, but that becomes messy as I have to pass each endpoint to the function in the client repeatedly. 

**Additional context**
See above.

**Comment by MikeWooster** [comment]
relates to #73

**Comment by MikeWooster** [comment]
fixed in #73
